janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -4520,6 +4521,85 @@ public void testMultiQueryMetricsWhenReadingFromBackend() {
         assertTrue(queryProfilerAnnotationIsPresent(t, QueryProfiler.MULTIQUERY_ANNOTATION));
     }
 
+    @Test
+    public void testLimitBatchSizeForMultiQuery() {
```

Comment:
Do you mean I should copy or move the test to `JanusGraphIndexTest` and add an index to one of the properties?

Reply:
Having thought about it for a while, I don't think your concern only applies to this feature. It's worth a general consideration whether features which are independent of index backends should be tested against different index backends anyway.
That, however, would mean that _every_ test should run on different indexes by default.

Reply:
I was just thinking if this change, in particular, can cause problems when using mixed indexes. Probably not...
One thing we could do before/after this PR is merged, is to run CI once with this feature enabled by default.

Reply:
Maybe I just found a bug by doing so. If I turn on `query.batch`, then BerkeleyElasticsearchTest::testExistsQuery succeeds on master but fails on your `feature/max-batch-size` branch.

Reply:
Can you elaborate what you tested exactly? I am not able to reproduce this even if I add the lines
```java
clopen(option(USE_MULTIQUERY), true);
clopen(option(LIMIT_BATCH_SIZE), true);
```
or
```java
clopen(option(USE_MULTIQUERY), true);
clopen(option(LIMIT_BATCH_SIZE), true);
```
at the beginning of `testExistsQuery`.

Reply:
Adding `clopen(option(USE_MULTIQUERY), true);` to the beginning of `testExistsQuery` does not make a difference, because that test has a few `clopen` in the middle. You need to change each `clopen`, e.g.

```java
clopen(option(FORCE_INDEX_USAGE), true);
```

into

```java
clopen(option(FORCE_INDEX_USAGE), true, option(USE_MULTIQUERY), true);
```

Reply:
I pushed the fix and also set `USE_MULTIQUERY` to `true` by default to trigger a CI run which could help detect similar problems, just as you suggested. I will to the same with `LIMIT_BATCH_SIZE` next before setting both back to `false`.